tests/PHPStan/Analyser/data/class-reflection-constants.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace ClassReflectionConstants;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld {
+	const A_CONSTANT = 'Constant';
+	const A_ARRAY = ['Constant', 987];
+
+	public const HELLO = 'Hello';
+	protected const WORLD = 'World';
+	private const SECRET = 123;
+}
+
+class Foo {
+    public function constantReflection(string $s) {
+        $cr = new \ReflectionClass(HelloWorld::class);
+        assertType("'Constant'", $cr->getConstant('A_CONSTANT'));
+        assertType("array{'Constant', 987}", $cr->getConstant('A_ARRAY'));
+        assertType("'Hello'", $cr->getConstant('HELLO'));
+        assertType("'World'", $cr->getConstant('WORLD'));
+        assertType('123', $cr->getConstant('SECRET'));
```

Comment:
this example illustrates the kind-of-stateful api the `ReflectionClass` has.

I think it would be usefull if phpstan would provide a `LocalVariableReflection` or similar kind of API, which allows to solve this stateful-apis at least for such local-variable cases. IMO using `ReflectionClass` in such a way is very common. most of the time it will not be used across classes (e.g. via properties)  etc. 

same is true for e.g. the stateful curl-api. in case phpstan would at a minimum support reflection constructor arguments based on a `PhpParser\Node\Expr\Variable` would really help